posthog/api/dashboards/dashboard.py

```diff
@@ -130,6 +130,20 @@ def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Dashboard:
         validated_data.pop("delete_insights", None)  # not used during creation
         validated_data = self._update_creation_mode(validated_data, use_template, use_dashboard)
         tags = validated_data.pop("tags", None)  # tags are created separately below as global tag relationships
+
+        if use_dashboard is not None:
+            # check permissions before creating dashboard
+            try:
+                existing_dashboard: Dashboard = Dashboard.objects.get(id=use_dashboard, team=team)
+            except Dashboard.DoesNotExist:
+                raise serializers.ValidationError({"use_dashboard": "Invalid value provided"})
+
+            if (
+                existing_dashboard.get_effective_privilege_level(self.context["request"].user.id)
+                == Dashboard.PrivilegeLevel.CAN_VIEW
+            ):
+                raise PermissionDenied(f"You don't have permission to add insights to dashboard: {use_dashboard}")
```

Comment:
To be in line with `insight.py`:
```suggestion
                raise PermissionDenied(f"You don't have permission to add insights to dashboard: {existing_dashboard.name}")
```